### PR TITLE
sets up drive base (ready for input)

### PR DESCRIPTION
**Pull request by echo2334**

### CODE REVIEW

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java
Comment (Elised9) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java:1]: put the first curly braket on the line below

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java [hunk 1]
```diff
@@ -17,6 +22,35 @@ public DriverController()
         
     }
 
+    public double getStick(ButtonMap stickAxis) 
+    {  if (this.xboxController != null)
```
Comment (Elised9) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java:26]: format this to coding standards

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java [hunk 2]
```diff
@@ -17,6 +22,35 @@ public DriverController()
...
+                    return xboxController.getRawAxis(1);
+                case XboxRIGHTSTICKX:
+                    return xboxController.getRawAxis(4); 
+                case XboxRIGHTSTICKY:
+                    return xboxController.getRawAxis(5); 
+                default:
+                    return 0;
+            }
+        }  
+        catch (Exception AxisNotFound)
```
Comment (Elised9) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java:42]: move catch by pressing tab

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java [hunk 3]
```diff
@@ -17,6 +22,35 @@ public DriverController()
...
+                    return 0;
+            }
+        }  
+        catch (Exception AxisNotFound) 
+        {
+            SmartDashboard.putString("ControllerError", "AxisNotFound");
+            return 0;
+        }
+        } 
+        else
```
Comment (Elised9) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java:48]: same with catch

Comment (Elised9) [reply TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java:48]: else

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java
Comment (dalittlet) [reply TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java:1]: Agreed ~Taylor

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java
Comment (Elised9) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java:1]: Same with this file

**Review by Elised9 — APPROVED** [review]

File: TitanRobotics2024/src/main/java/frc/robot/ExternalLibraries/LimelightHelpers.java [hunk 4]
```diff
@@ -0,0 +1,774 @@
...
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonFormat.Shape;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class LimelightHelpers {
+
+    public static class LimelightTarget_Retro {
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/ExternalLibraries/LimelightHelpers.java:29]: This class does not follow coding standards. Nor should this be in the drive base branch at this time. ~Taylor

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/AprilTagTargeting.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+import frc.robot.ExternalLibraries.LimelightHelpers;
+
+public class AprilTagTargeting implements Subsystem //This class contains functions for finding and locking onto elements of the field using their April Tags.
+{
+    private static AprilTagTargeting instance = null;
+
+    public static AprilTagTargeting getInstance() 
+    {
+        if (instance == null) {
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/AprilTagTargeting.java:11]: Does not follow coding standards

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/AprilTagTargeting.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+            }
+        }
+
+        else
+        {
+            SourceFound = false;
+        }
+    }
+
+    public void findEnemySpeaker() //A defensive function that looks for the opponent's speaker and reacts when it is found. Use as a last resort only. IDs are 7,8 for blue(targeted as red), 3,4 for red(targeted as blue).
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/AprilTagTargeting.java:111]: We should NEVER need this function, let's go ahead and remove it

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java:1]: This entire class doesn't follow coding standards

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java [hunk 7]
```diff
@@ -18,20 +17,16 @@ public void update() {
...
-    
-     private int portNumber;
+
+    private int portNumber;
     private final MotorController motor;
-    
-    public ModifiedMotors(int portNumber) 
-    {
+
+    public ModifiedMotors(int portNumber) {
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java:24]: Needs to follow coding standards

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/Ramp.java
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/Ramp.java:1]: This class needs to be removed from this PR, the ramp is not part of the drive base

**Comment by thebuilder6** [comment]
Pull request outdated by pull request #3 Driving Robot 
All significent changes have been substantially altered

**Pull request merged**